target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_minor.c

```diff
@@ -61,29 +63,46 @@ static int mtdsplit_parse_minor(struct mtd_info *master,
 
 	hdr_len = sizeof(hdr);
 	err = mtd_read(master, 0, hdr_len, &retlen, (void *) &hdr);
-	if (err)
+	if (err) {
+		pr_err("MiNOR mtd_read error: %d\n", err);
 		return err;
+	}
 
-	if (retlen != hdr_len)
+	if (retlen != hdr_len) {
+		pr_err("MiNOR mtd_read too short\n");
 		return -EIO;
+	}
 
 	/* match header */
-	if (hdr.yaffs_type != YAFFS_OBJECT_TYPE_FILE)
-		return -EINVAL;
-
-	if (hdr.yaffs_obj_id != YAFFS_OBJECTID_ROOT)
-		return -EINVAL;
-
-	if (hdr.yaffs_sum_unused != YAFFS_SUM_UNUSED)
-		return -EINVAL;
-
-	if (memcmp(hdr.yaffs_name, YAFFS_NAME, sizeof(YAFFS_NAME)))
-		return -EINVAL;
+	if (hdr.yaffs_type != YAFFS_OBJECT_TYPE_FILE) {
+		pr_info("MiNOR YAFFS first type not matched\n");
+		return 0;
+	}
+
+	if (hdr.yaffs_obj_id != YAFFS_OBJECTID_ROOT) {
+		pr_info("MiNOR YAFFS first objectid not matched\n");
+		return 0;
+	}
+
+	if (hdr.yaffs_sum_unused != YAFFS_SUM_UNUSED) {
+		pr_info("MiNOR YAFFS first sum not matched\n");
+		return 0;
+	}
+
+	if ((memcmp(hdr.yaffs_name, YAFFS_NAME_KERNEL,
+					sizeof(YAFFS_NAME_KERNEL)))
+			&& (memcmp(hdr.yaffs_name, YAFFS_NAME_BOOTIMAGE,
+					sizeof(YAFFS_NAME_BOOTIMAGE)))) {
```

Comment:
a very minor nit: the wrapping seems a little excessive to me, the kernel has deprecated 80-col wrapping some 4 years ago now :) and I would also align the checks differently to highlight that both `memcmp()` happen at the same logical level.

So I'd write this like so:
```
        if ((memcmp(hdr.yaffs_name, YAFFS_NAME_KERNEL, sizeof(YAFFS_NAME_KERNEL))) &&
            (memcmp(hdr.yaffs_name, YAFFS_NAME_BOOTIMAGE, sizeof(YAFFS_NAME_BOOTIMAGE)))) {
```

That aside, this LGTM